AnnoDesigner/Localization/TranslateExtension.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Windows.Data;
+
+namespace AnnoDesigner.Localization
+{
+    public class TranslateKeyConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            var translations = value as IDictionary<string, string>;
+
+            if (parameter is string key && translations.TryGetValue(key, out var translation))
+            {
+                return translation;
+            }
+#if DEBUG
+            throw new Exception($"Missing translation or not string translation key \"{parameter}\"");
+#else
+            return $"!{parameter}";
+#endif
```

Comment:
Its just occurred to me that this could still error in Release if `parameter` is null, we may want to replace this with `{parameter ?? "!<some placeholder value>"}` - <some placeholder value> could be constant defined somewhere.

Reply:
When formatting a null, empty string is returned.